forte/processors/data_augment/algorithms/text_replacement_op.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Class for data augmentation algorithm. The text replacement op will
+replace a piece of text with data augmentation algorithms.
+"""
+from typing import Dict, Any
+from abc import abstractmethod, ABC
+from forte.data.ontology.core import Entry
+
+__all__ = [
+    "TextReplacementOp",
+]
+
+
+class TextReplacementOp(ABC):
+    r"""
+    The base class holds the data augmentation algorithm.
+    We leave the :func: replace method to be implemented
+    by subclasses.
+    """
+    def __init__(self, configs: Dict[str, Any]):
```

Comment:
Let's use the `Config` object in Forte (which is the Hparam object in Texar) to keep interface slightly more consistent.